### DISCUSSION TITLE
Using /files with custom role permissions

### DISCUSSION THREAD

**Discussion by justrealmilk** [directus/directus#7264] [Archive]
As per https://github.com/directus/directus/discussions/5248, I'm trying to limit public /files to specific folders.

It half works in that it returns a list of file IDs.

/files response:
```json
{
  "data": [
    "033ab914-85a8-4966-87ca-1c7b8541ccb1",
    "08496b1a-40f8-47e3-aa73-81f57c5de3c5",
    "08c62def-6de4-48eb-bebd-2c1de06ee7a8"
  ]
}
```

I've tried a few things but can't get more than this.

![image](https://user-images.githubusercontent.com/156681/128585499-21161dc5-f6b1-4de7-bd03-e82f697ed5bb.png)
![image](https://user-images.githubusercontent.com/156681/128585505-0135198f-0706-47b4-ae52-23dc738b0c36.png)

There was a severe SQL error prior to making Folders available to the role (which I guess makes sense).

**Comment by rijkvanzanten** [comment]
You'll also have to specify what fields under files are available to your role. I think it's currently set to none, which will only retrieve the primary key. See the next tab "Field Permissions"

**Reply by justrealmilk** [reply]
Well I'll say. Bloody brilliant. Thanks again, @rijkvanzanten
